presto-docs/src/main/sphinx/admin/tuning.rst

```diff
@@ -10,29 +10,392 @@ Config Properties
++                                          +                                          +                                    +http requests. Presto server sends update of query     +
++                                          +                                          +                                    +status whenever it is different then the one that      +
++                                          +                                          +                                    +client knows about. However in order to ensure client  +
++                                          +                                          +                                    +that connection is still alive, server sends this data +
++                                          +                                          +                                    +after delay declared internally in HTTP headers (by    +
++                                          +                                          +                                    +default ``200 ms``). This property tells how many      +
++                                          +                                          +                                    +threads are designated to handle this delay. If the    +
++                                          +                                          +                                    +property turn out to low it's possible that the update +
++                                          +                                          +                                    +time will increase even significantly when comparing   +
++                                          +                                          +                                    +to requested value (``200ms``). Increasing this value  +
++                                          +                                          +                                    +may solve the problem, but it generate a cost of       +
++                                          +                                          +                                    +additional memory even if threads are not used all the +
++                                          +                                          +                                    +time. If there is no problem with updating status of   +
++                                          +                                          +                                    +query this value should not be manipulated.            +
++------------------------------------------+------------------------------------------+------------------------------------+-------------------------------------------------------+
++        ``task.max-index-memory``         +          ``String`` (data size)          +             ``64 MB``              +Max size of index cache in memory used for index based +
```

Comment:
Index join is presto optimization based on IndexJoinOptimizer. It's used when index is available for at least one of joined tables has index available for JOIN condition. It's available for each connector providing indexes then.